jupyter_client/kernelspecapp.py

```diff
@@ -337,5 +369,20 @@ def start(self) -> None:
             return self.subapp.start()
 
 
+def _limit_to_missing(
+    paths: dict[str, str], specs: dict[str, t.Any]
+) -> tuple[dict[str, str], dict[str, t.Any]]:
+    specs_: dict[str, t.Any] = {
+        k: v
+        for k, v in specs.items()
+        # If have kernel installed from current environment
+        # Probably not the best way to do this, but it works in edge cases I've run into.
+        if (prog := v["spec"]["argv"][0]) != "python" and not Path(prog).exists()
```

Comment:
This should use `shutil.which` to resolve things on $PATH, since it is not a requirement that `argv[0]` is an absolute path.